src/backend/access/transam/slru.c

```diff
@@ -762,7 +762,12 @@ SimpleLruDownloadSegment(SlruCtl ctl, int pageno, char const* path)
 	}
 	segno = pageno / SLRU_PAGES_PER_SEGMENT;
 
-	buffer = palloc(BLCKSZ * SLRU_PAGES_PER_SEGMENT);
+	if (!strcmp(GetConfigOption("neon.use_communicator_worker", true, false), "on")) {
```

Comment:
Just making sure this isn't a thinko: If neon's communicator worker is NOT enabled (`if (!strcmp(..., "on"))`) we don't alloc a buffer, and when it is enabled (`else`) we do alloc the buffer?

Reply:
!strcmp(..., "on") returns `true` when the strings are equal. Meaning, the condition `if (!strcmp(GetConfigOption("neon.use_communicator_worker", true, false), "on"))` will be `true` when the setting is `"on"`.